src/or_trajopt/planning.py

```diff
@@ -560,11 +565,8 @@ def PlanToTSR(self, robot, tsrchains, is_interactive=False, **kw_args):
         with robot.CreateRobotStateSaver(p.ActiveDOF):
             # Set robot DOFs to DOFs in optimization problem
             prob.SetRobotActiveDOFs()
-
             # Check that trajectory is collision free
-            from trajoptpy.check_traj import traj_is_safe
-            if not traj_is_safe(waypoints, robot):
-                return PlanningError("Result was in collision.")
+            self._checkCollisionForIKSolutions(robot, waypoints)
```

Comment:
You can call `util.GetCollisionCheckPts` to get a list of joint configurations. Then, for each configuration, you can perform whatever tests you want, e.g. collision, self-collision, constraint violation, etc.